### ISSUE TITLE
BUG: Seaborn failed to accept a Series produced by `xorbits.numpy`

### ISSUE THREAD

**Issue by onesuper** [xprobe-inc/xorbits#522]
### Describe the bug

After converting a column using `xorbits.numpy`, I get an error when passing the dataframe to seaborn for plotting, and the same code runs fine with `pandas` & `numpy`.

### To Reproduce
To help us to reproduce this bug, please provide information below:

1. Your Python version
3.11.3

2. The version of Xorbits you use
0.3.1

3. Versions of crucial packages, such as numpy, scipy and pandas
Seaborn: 0.12.2

4. Full stack of the error.
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[19], line 24
     21 # Plot the frequency using log coordinates
     22 df['log_text_length'] = np.log(df['text_length'])
---> 24 sns.histplot(data=df, x='log_text_length', bins=20, kde=True)

File [~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/seaborn/distributions.py:1395](https://file+.vscode-resource.vscode-cdn.net/Users/onesuper/Code/xorbits/gallery/~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/seaborn/distributions.py:1395), in histplot(data, x, y, hue, weights, stat, bins, binwidth, binrange, discrete, cumulative, common_bins, common_norm, multiple, element, fill, shrink, kde, kde_kws, line_kws, thresh, pthresh, pmax, cbar, cbar_ax, cbar_kws, palette, hue_order, hue_norm, color, log_scale, legend, ax, **kwargs)
   1374 def histplot(
   1375     data=None, *,
   1376     # Vector variables
   (...)
   1392     **kwargs,
   1393 ):
-> 1395     p = _DistributionPlotter(
   1396         data=data,
   1397         variables=_DistributionPlotter.get_semantics(locals())
   1398     )
   1400     p.map_hue(palette=palette, order=hue_order, norm=hue_norm)
   1402     if ax is None:

File [~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/seaborn/distributions.py:113](https://file+.vscode-resource.vscode-cdn.net/Users/onesuper/Code/xorbits/gallery/~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/seaborn/distributions.py:113), in _DistributionPlotter.__init__(self, data, variables)
    107 def __init__(
    108     self,
    109     data=None,
...
File [~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/xorbits/_mars/core/entity/core.py:141](https://file+.vscode-resource.vscode-cdn.net/Users/onesuper/Code/xorbits/gallery/~/opt/miniconda3/envs/xorbits_gallery/lib/python3.11/site-packages/xorbits/_mars/core/entity/core.py:141), in Entity.__getattr__(self, attr)
    140 def __getattr__(self, attr):
--> 141     return getattr(self._data, attr)

AttributeError: 'SeriesData' object has no attribute '__array__'
```

6. Minimized code to reproduce the error.

```python
import xorbits.numpy as np
import xorbits.pandas as pd
#import numpy as np
import seaborn as sns
data = {
    'column1': ['value1', 'value2', 'value3'],
    'column2': ['value4', 'value5', 'value6'],
    'column3': ['value7', 'value8', 'value9']
}


df=pd.DataFrame(data)

from transformers import BertTokenizer

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')

# Create a new column with the number of tokens for each text
df['text_length'] = df['column1'].apply(lambda text: len(tokenizer.tokenize(text)))

# Plot the frequency using log coordinates
df['log_text_length'] = np.log(df['text_length'])

sns.histplot(data=df, x='log_text_length', bins=20, kde=True)
```

**Comment by qianduoduo0904** [comment]
I must have been addressed in #282 , still have some bug? @UranusSeven

**Comment by UranusSeven** [comment]
Looks like `__array__` is implemented only for dataframe in #282. It is also needed to implement it for series and index.

**Comment by UranusSeven** [comment]
I've created another 2 issues for series and index respectively:
 #527 #528
